portal.go

```diff
@@ -1012,6 +1012,12 @@ func (portal *Portal) HandleMatrixMessage(evt *event.Event) {
 	if portal.bridge.IM.Capabilities().RichLinks {
 		imessageRichLink = portal.convertURLPreviewToIMessage(evt)
 	}
+	var metadata *interface{}
```

Comment:
Pointers to interfaces shouldn't be used, interfaces can be nil on their own. Might also be better to require the data to be an object (`map[string]interface{}`) instead of any JSON type.